fdbserver/DataDistribution.actor.cpp

```diff
@@ -4020,26 +4031,19 @@ ACTOR Future<Void> perpetualStorageWiggler(AsyncVar<bool>* stopSignal,
 				self->includeStorageServersForWiggle();
 				TraceEvent("PerpetualStorageWigglePause", self->distributorId)
 				    .detail("ProcessId", pid)
+				    .detail("ExtraHealthyTeamCount", extraTeamCount)
+				    .detail("HealthyTeamCount", self->healthyTeamCount)
 				    .detail("StorageCount", movingCount);
 			} else {
-				// pre-check whether wiggling chosen servers still satisfy replica requirement
-				excludedServerIds.clear();
-				for (const auto& info : self->pid2server_info[self->wigglingPid.get()]) {
-					excludedServerIds.push_back(info->id);
-				}
-				if (_exclusionSafetyCheck(excludedServerIds, self)) {
```

Comment:
It seems like there would still be some value in doing the exclusion safety check, even with the pause threshold, right?
Since the threshold resets because DD restarts every time there is a recovery or configuration change, and also because you could avoid an exclude/re-include due to cluster health if the wiggle starts by excluding a server known to be required for cluster health?

Reply:
I think `_exclusionSafetyCheck` which check the replica>=1 is unnecessary here. The reason is in perpetual wiggle, shards will be deleted on the original team if and only if it's written to another team. Perpetual wiggle just changes the location of replications but not deletes them,  so the number of replications in the cluster keeps the same.
Another example is, if the replica mode is single, we still can do perpetual wiggle. But with this check, that will never happen.